test/conftest.py

```diff
@@ -0,0 +1,303 @@
+        softhsm2_mod = "/usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so"
+    else:
+        softhsm2_mod = "/usr/lib/softhsm/libsofthsm2.so"
+    if not os.path.exists(softhsm2_mod):
+        pytest.skip("libsofthsm2.so not available on system")
+
+    prepare_softhsm2(tmp_path_factory.mktemp('blub'), softhsm2_mod)
+
+
+# If running under qemu use the prepared system bus otherwise start a
+# dedicated session bus
+@pytest.fixture(scope='session')
+def select_system_or_session_bus():
+    if not have_service():
+        pytest.skip("No service")
+
+    with open("/proc/cmdline") as f:
+        if re.search("init=[^ ]*/qemu-test-init", f.read()):
+            print("Running from QEMU")
+        else:
+            print("Running locally")
+
+    import subprocess
+    import os
+
+    try:
+        # Run the dbus-launch command and capture its output
+        output = subprocess.check_output(["dbus-launch", "--sh-syntax"],
+                                         universal_newlines=True,
+                                         shell=True)
+
+        # Parse the output to extract environment variable assignments
+        env_vars = {}
+        for line in output.splitlines():
+            if "=" in line:
+                key, value = line.split("=", 1)
+                env_vars[key] = value
+
+        # Set the environment variables in the current process's environment
+        for key, value in env_vars.items():
+            os.environ[key] = value
+
+        # Now you can access the environment variables in your Python script
+        dbus_session_bus_address = os.environ.get("DBUS_SESSION_BUS_ADDRESS")
+        if dbus_session_bus_address:
+            print(f"DBUS_SESSION_BUS_ADDRESS: {dbus_session_bus_address}")
+
+    except subprocess.CalledProcessError as e:
+        print(f"Error running dbus-launch: {e}")
+    except Exception as e:
+        print(f"An error occurred: {e}")
+
+    yield
+
+    pid = os.environ["DBUS_SESSION_BUS_PID"]
+    print(f"Killing PID {pid}")
+    run(f"kill {pid}")
+
+
+def have_service():
+    return string_in_config_h("ENABLE_SERVICE 1")
+
+
+@pytest.fixture
+def rauc_service(select_system_or_session_bus):
+    pass
+
+
+def create_system_files():
+    shutil.rmtree("images", ignore_errors=True)
+    os.mkdir("images")
+    open("images/rootfs-0", mode="w").close()
+    open("images/rootfs-1", mode="w").close()
+    open("images/appfs-0", mode="w").close()
+    open("images/appfs-1", mode="w").close()
+
+
+def _rauc_dbus_service(tmp_path, conf_file, bootslot):
+
+    service = subprocess.Popen(f'rauc service --conf={conf_file} '
+                               f'--mount={tmp_path}/mnt '
+                               f'--override-boot-slot={bootslot}'.split())
+
+    bus = SessionBus()
+
+    # Wait for de.pengutronix.rauc to appear on the bus
+    timeout = time.monotonic() + 5.0
+    while True:
+        try:
+            bus.get("de.pengutronix.rauc", "/")
+            break
+        except GLib.GError:
```

Comment:
Do you want to keep pydbus for other tests that would be added later? If it's just for checking if the service is started, I'd prefer to avoid the dependency.

Reply:
I've kept it first of all and have a prepared example on how to use it for dbus tests: https://github.com/ejoerns/rauc/commit/aa33a7ef08a34782c523d281898d3bf8270c5ca1